mongoo/mongoo.go

```diff
@@ -0,0 +1,106 @@
+package mongoo
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/faabiosr/cachego"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type (
+	mongoo struct {
+		collection *mongo.Collection
+	}
+
+	mongooContent struct {
+		Duration int64
+		Key      string `bson:"_id"`
+		Value    string
+	}
+)
+
+// New creates an instance of Mongo cache driver
+func New(collection *mongo.Collection) cachego.Cache {
+	return &mongoo{collection}
+}
+
+func (m *mongoo) Contains(key string) bool {
+	_, err := m.Fetch(key)
+	return err == nil
+}
+
+// Delete the cached key from Mongo storage
+func (m *mongoo) Delete(key string) error {
+	_, err := m.collection.DeleteOne(context.TODO(), bson.M{"_id": bson.M{"$eq": key}})
+	return err
+}
+
+// Fetch retrieves the cached value from key of the Mongo storage
+func (m *mongoo) Fetch(key string) (string, error) {
+	content := &mongooContent{}
+
+	if result := m.collection.FindOne(context.TODO(), bson.M{"_id": bson.M{"$eq": key}}); result != nil {
+		if result.Err() != nil {
+			return "", result.Err()
+		}
+		err := result.Decode(&content)
+		if err != nil {
+			return "", err
+		}
+		if content.Duration == 0 {
+			return content.Value, nil
+		}
+
+		if content.Duration <= time.Now().Unix() {
+			_ = m.Delete(key)
+			return "", errors.New("cache expired")
+		}
+		return content.Value, nil
+	}
+
+	return "", errors.New("cache expired")
+}
+
+func (m *mongoo) FetchMulti(keys []string) map[string]string {
+	result := make(map[string]string)
+
+	cur, err := m.collection.Find(context.TODO(), bson.M{"_id": bson.M{"$in": keys}})
+	if err != nil {
+		return result
+	}
+	content := &mongooContent{}
+	defer cur.Close(context.Background())
```

Comment:
`Defer` statement should be called right after the `if`, living right now doesn't be a harm, however will improve the readability.

Reply:
Thanks for the suggestion. Make sense. I'll  make changes and push.